composition-js/src/validate.ts

```diff
@@ -257,6 +277,19 @@ function initialSubgraphPaths(kind: SchemaRootKind, subgraphs: QueryGraph): Root
   return subgraphs.outEdges(root).map(e => initialState.add(subgraphEnteringTransition, e, noConditionsResolution));
 }
 
+function possibleRuntimeTypeNamesSorted(path: RootPath<Transition>): string[] {
+  const types = path.tailPossibleRuntimeTypes().map((o) => o.name);
```

Comment:
Is this already deduped?

Reply:
It is. And it does take a small effort to make sure so I commented as much on the method. But essentially, `tailPossibleRuntimeTypes` in `GraphPath` is only updated through the (private) `updateRuntimeTypes` method in `graphPaths.ts`, and the only place we "push" there is guarded by I first check to ensure the element is not already there.